### PR TITLE
Remove Vue Select and Add Column Filter Slot

### PR DESCRIPTION
**Pull request by robgaston1**
Addresses issue https://github.com/xaksis/vue-good-table/issues/675

**Notes:**
* Add a slot in `VgtFilterRow`
* Create a method `updateSlotFilter` that includes the custom filter with the rest of the column filter values
* Expose the method as a prop on the slot so it can be optionally used
* Allow a `slotFilterField` to be passed in, because the property used for display may not be the same as the property for filtering
* Remove the `vue-select` code to reduce the overall bundle size.

### CODE REVIEW

**Comment by xaksis** [comment]
@robgaston1 could you please add documentation for this as well? That would be immensely helpful.

-  how to use the new slot
-  a guide to migration for people who're currently using v-select.

File: vp-docs/guide/advanced/README.md [hunk 1]
```diff
@@ -83,6 +83,114 @@ Sometimes you might want to customize column headers. You can do that in the fol
...
+{
+  label: 'Name',
+  field: 'name.displayName',
+  filterOptions: {
+    customFilter: true,
+    slotFilterField: 'name.id'
+  }
+}
+```
+::: tip Upgrading from versions 2.19.0-2.20.0
```
Comment (robgaston1) [inline vp-docs/guide/advanced/README.md:148]: @xaksis I wasn't sure where it was desirable to have the migration guide. I notice there are some things like that in the wiki, but wasn't sure if I can/should contribute to that. 

I just put it here for now, but let me know if different.

File: src/components/VgtFilterRow.vue [hunk 2]
```diff
@@ -162,17 +145,25 @@ export default {
       this.updateFilters(column, value);
     },
 
+    updateSlotFilter(column, value) {
+      let fieldToFilter = column.filterOptions.slotFilterField || column.field;
+      if (typeof column.filterOptions.formatValue === 'function') {
```
Comment (robgaston1) [inline src/components/VgtFilterRow.vue:150]: This was an addition today, FYI. In working with this branch in our team project, I was formatting the filter data right before passing it in, but passing the function into the table made it much cleaner and enhanced reusability in our own codebase.

File: dev/App.vue [hunk 3]
```diff
@@ -202,8 +175,7 @@ export default {
           createdAt: '',
           score: 0.03343,
           bool: null,
-          exact: 'rematch',
-          multiselect: 'hello world',
+          exact: 'rematch'
```
Comment (TheJaredWilcurt) [inline dev/App.vue:178]:
```suggestion
          exact: 'rematch',
```

File: vp-docs/guide/advanced/README.md [hunk 4]
```diff
@@ -83,6 +83,129 @@ Sometimes you might want to customize column headers. You can do that in the fol
...
+/>
+```
+``` javascript
+// vue-select emits an array of any objects selected in the dropdown
+// which is being converted to a string of ids to pass into the column filter value
+data: {
+  optionList: [
+    {
+       name: 'Joan',
+       id: 1
```
Comment (TheJaredWilcurt) [inline vp-docs/guide/advanced/README.md:184]:
```suggestion
      name: 'Joan',
      id: 1
```

File: vp-docs/guide/advanced/README.md [hunk 5]
```diff
@@ -83,6 +83,129 @@ Sometimes you might want to customize column headers. You can do that in the fol
...
+You can add a function to handle the filtering logic in your own component, or optionally `updateFilters` can be used. 
+The `updateFilters` method is in `vue-good-table` and will include your custom filter value with the other column filters. 
+You can also provide a function to `formatValue` inside of `filterOptions` to transform the value before filtering on it. 
+```html
+<vue-good-table
+  :columns="columns"
+  :rows="rows">
+  <template slot="column-filter" slot-scope="{ column, updateFilters }">
+    <my-custom-filter
+      v-if="column.filterOptions.customFilter"
```
Comment (TheJaredWilcurt) [inline vp-docs/guide/advanced/README.md:132]:
```suggestion
      v-if="column.filterOptions.customFilter"
```

File: vp-docs/guide/advanced/README.md [hunk 6]
```diff
@@ -83,6 +83,129 @@ Sometimes you might want to customize column headers. You can do that in the fol
...
+      id: 2
+    }
+  ],
+  columns: [
+    {
+      label: 'name',
+      field: 'name',
+      filterOptions: {
+        customFilter: true,
+        formatValue: this.formatFilterValue
```
Comment (TheJaredWilcurt) [inline vp-docs/guide/advanced/README.md:197]:
```suggestion
        formatValue: this.formatFilterValue
      }
```

File: vp-docs/guide/advanced/README.md [hunk 7]
```diff
@@ -83,6 +83,129 @@ Sometimes you might want to customize column headers. You can do that in the fol
...
+```
+* Make sure to set the `multiple` attribute for a multiselect filter.
+* Set an array of options on the `options` attribute of `v-select`. If you were using the 
+built in multiselect filter, move them from the column property `filterOptions.multiSelectDropdownItems`.
+* `vue-select` emits an array of values when set to `multiple`. To convert the array of data into 
+a comma separated string or another format, provide a function on `filterOptions.formatValue`.  
+``` html
+<v-select
+  :options="optionList"
+  multiple
```
Comment (TheJaredWilcurt) [inline vp-docs/guide/advanced/README.md:173]:
```suggestion
  label="name"
  multiple
```

File: vp-docs/guide/advanced/README.md [hunk 8]
```diff
@@ -83,6 +83,129 @@ Sometimes you might want to customize column headers. You can do that in the fol
...
+      field: 'name',
+      filterOptions: {
+        customFilter: true,
+        formatValue: this.formatFilterValue
+    }
+  ]
+},
+methods: {
+  formatFilterValue(valuesArray) {
+    return values.map((value) => value.id).join(',');
```
Comment (TheJaredWilcurt) [inline vp-docs/guide/advanced/README.md:203]:
```suggestion
    return valuesArray.map((value) => value.id).join(',');
```

**Comment by TheJaredWilcurt** [comment]
I only had a few PR comments, and they have all been resolved.

I am able to run this branch locally and follow the documentation to migrate to manually adding vue-select into the new custom-filters slot. This seems to work as intended. Will wait on @xaksis to approve.

Once merged the build size will be reduced by ~15% due to the removal of Vue-Select.

Reduction in build size


If merged this will resolve several other issues:

* Fixes #724
* Fixes #714
* Fixes #708
* Fixes #678
* Fixes #675
* Fixes #655
* Fixes #616

It will also effect this one #732. Debounce will need to be removed on custom filter slots

**Comment by xaksis** [comment]
@robgaston1 Fantastic! Thank you for this PR. @TheJaredWilcurt thank you for testing. So happy about the bundle size reduction. 👍  PR merged and released.

**Comment by TheJaredWilcurt** [comment]
All related issues and their stakeholders have been updated. All 7 "fixed by" issues closed.

**Pull request merged**